### DISCUSSION TITLE
AMS filament drop-down filtering for Bambu printers

### DISCUSSION THREAD

**Discussion by shoka1ei** [OrcaSlicer/OrcaSlicer#7765] [Ideas]
### The problem
---
If you're using the Bambu Lab's AMS with custom profiles, and have to map the custom profile to the AMS slot (because you inserted a new spool, for example), you have to scroll down through a lot of generic/built-in/default profiles to find yours.
Doing so may not take a lot of time but can be frustrating.

On the screenshot bellow you can find "Bambu" (default profiles) and "Bambu Lab" (custom profiles) that are mixed together, making them hard to spot if you're not paying much attention

![image](https://github.com/user-attachments/assets/47e0d0d9-f35d-4d51-8b44-e8230ae12716)

### The solution
---
I believe there are 2 solutions that would solve the issue:

- An option to filter out unwanted profiles, just like in the System Filaments tab (screenshot)

![image](https://github.com/user-attachments/assets/46d8dd3d-d6a1-4ab5-a51c-943f937aecee)

- Separation of custom profiles in the drop-down menu (as in the screenshot), and moving them to the top of the list for easier navigation

![image](https://github.com/user-attachments/assets/406b7c6b-08bd-4676-8729-0dfc97e6a3b1)

**Comment by sstickel** [comment]
To me this dropdown list is extremely annoying. Why the heck would I want to be able to select a filament brand/type I don't even have a profile for?

When initially setting up Orca I selected my machine and only added a handful of filaments. (Generic PLA/PETG/ASA and a few Bambu ones) Based on these initial set I created my custom filament profiles.


Yet the list in the Network Plugin / AMS still shows me a full blown list of clutter containing filaments I haven't even heard about.

Subscribing this as hopefully someone comes up with an idea how to solve this. But I have only little hope to get this solved because afaik the Network Plugin which creates this list is nailed shut and closed source.

**Reply by shoka1ei** [reply]
Hey, my current workaround is using the minus sign (-) before the profile name. In ASCII table it's before any letter, so all custom profiles stay on top
<img width="407" height="499" alt="image" src="https://github.com/user-attachments/assets/ca6f191c-8406-4928-ab34-4d5487ada936" />

**Comment by sstickel** [comment]
Well, it's so obvious that I have overseen that one. Thx!
